### PR TITLE
Added documentation for imdecode

### PR DESCRIPTION
**Pull request by gurumurthys**
Added documentation for imdecode. Removed one of the parameters "out" as it was not used. The code crashes when flag=0 and to_rgb is set to 1 (default) as it internally tries to convert the image into opencv format using cvtColor. Hence,  I have added documentation that specifies to_rgb to be set to 0 when flag = 0. @mli, @zackchase, @madjam, @nswamy

### CODE REVIEW

File: python/mxnet/image.py [hunk 1]
```diff
@@ -37,8 +37,8 @@ def imdecode(buf, **kwargs):
         1 for three channel color output. 0 for grayscale output.
     to_rgb : int, optional, default=1
         1 for RGB formatted output (MXNet default). 0 for BGR formatted output (OpenCV default).
-    out : NDArray, optional
-        Output buffer. Use `None` for automatic allocation.
+
+    # removed out parameter as it is not used
```
Comment (piiswrong) [inline python/mxnet/image.py]: out is used.

**Comment by piiswrong** [comment]
the crash is a bug and we should fix the code instead of doc

File: python/mxnet/image.py [hunk 2]
```diff
@@ -37,8 +37,8 @@ def imdecode(buf, **kwargs):
...
 
     Returns
     -------
```
Comment (zackchase) [inline python/mxnet/image.py]: Where is the description of the returned values?

Comment (gurumurthys) [inline python/mxnet/image.py]: I have fixed this at https://github.com/dmlc/mxnet/pull/6134 

Will update the doc once that is pushed

File: python/mxnet/image.py (hunk 1 above)
Comment (gurumurthys) [inline python/mxnet/image.py]: Can you please point out where it is used? I am not able to find the relevant code in image_io.cc

Comment (gurumurthys) [inline python/mxnet/image.py]: I am unable to figure out how out is used to pass the output image. Can you point to the location of where it is used?

Comment (mli) [inline python/mxnet/image.py]: it is used by _cvimdecode

**Comment by gurumurthys** [comment]
Fixed the doc

File: python/mxnet/image.py (hunk 2 above)
Comment (pracheer) [inline python/mxnet/image.py]: @gurumurthys Looks like #6134  has been pushed. Can you update the code now?

**Pull request merged**